### ISSUE TITLE
Converting circular structure to JSON error

### ISSUE THREAD

**Issue by c9482** [volkanceylan/Serenity#3962]
I have the following code which generates "Uncaught TypeError: Converting circular structure to JSON". Many thanks in in advance for any pointers on how to avoid this error.

    @Serenity.Decorators.registerClass()
    export class RequestAttachmentDialog extends AttachmentDialog
    {

        constructor()
        {
            super();
        }

        loadEntity(entity: AttachmentRow) {
            super.loadEntity(entity);

            // Type is only editable for new rows
            Serenity.EditorUtils.setReadOnly(this.form.Type, !this.isNew());
            Serenity.EditorUtils.setReadOnly(this.form.RequestId, true);

            Default.RequestService.RetrieveStatus({
                EntityId: this.form.RequestId
            }, response => {
                if (response.Status === "Issued" || response.Status === "Denied") {
                    this.element.find(".Location").hide();
                }
            }, { async: false });
        }

    }

![image](https://user-images.githubusercontent.com/17013052/46171918-94488200-c267-11e8-9be1-da1838a24860.png)

**Comment by minhhungit** [comment]
can you show code about `AttachmentDialog`

**Comment by c9482** [comment]
namespace LCPermits.Default {

    @Serenity.Decorators.registerClass()
    @Serenity.Decorators.responsive()
    @Serenity.Decorators.maximizable()
    export class AttachmentDialog extends Serenity.EntityDialog<AttachmentRow, any> {
        protected getFormKey() { return AttachmentForm.formKey; }
        protected getIdProperty() { return AttachmentRow.idProperty; }
        protected getLocalTextPrefix() { return AttachmentRow.localTextPrefix; }
        protected getNameProperty() { return AttachmentRow.nameProperty; }
        protected getService() { return AttachmentService.baseUrl; }

        protected form = new AttachmentForm(this.idPrefix);

        constructor()
        {
            super();

            this.form.Type.changeSelect2(e =>
            {
                var type = Q.toId(this.form.Type.value);
                if (type != null)
                {
                    this.form.Description.value = AttachmentTypeRow.getLookup().itemById[type].Description;
                }
            });
        }

        dialogOpen() {
                    super.dialogOpen();
                   this.element.closest(".ui-dialog").find(".ui-icon-maximize-window").click();
        }

    }

**Comment by minhhungit** [comment]
unlucky I can not reproduce the issue, sorry

**Comment by c9482** [comment]
No problem. Originally, I used Default.RequestService.Retrieve method which returned RequestRow (which contains an array of AttachmentRow - possible circular reference here?). I thought this was the problem so I created Default.RequestService.RetrieveStatus method which only returns Status field but it did not help. I've ran into this several times and it would be nice to know why I am getting this error. In the meantime, I devised a workaround. Thanks again for looking at this problem, @minhhungit .
